### DISCUSSION TITLE
Video files of test runs are attached to the Allure Report as 0KB files or multiples of 256KB in size

### DISCUSSION THREAD

**Discussion by lukaszewskimichal** [allure-framework/allure2#2828] [Questions & Support] [answered]
Hi All,

I use Allure Report to present the results of e2e tests written using Playwright and PyTest.
I have tried to implement adding test run video to the test report in Allure Report but I have encountered a issue where files added to Allure Report are empty (0 KB in size) even though the source video files are valid, alternatively the files are added with a fixed size of 256 KB (or a multiple of this size) no matter what size the source video files are.
I add the video files to Allure Report using the allure.attach.file() method in WEBM format.

![image](https://github.com/user-attachments/assets/ad3f98b6-caac-4ecb-b607-6ceb4d9e5d41)

I have tried to find a solution to this problem on the Internet, but without success, so I decided to ask for advice here.
I only found a thread about this problem on stackoverflow, but without a solution ([link to this thread](https://stackoverflow.com/questions/78096625/allure-report-attaches-video-with-size-zero-in-playwright-tests)).

I would appreciate your help or any guidance.

**Comment by baev** [comment, marked as answer]
The issue is with file buffers. The file system buffers the data on write, and if you call `allure.attach.file()` method before the data is flushed to disk — you'll get the empty attachment. It seems like the file system flushes data in 256K chunks — that's why you see files of such size.

You can try to use `fsync` to force buffers to be flushed before calling `allure.attach.file()` method.

**Reply by lukaszewskimichal** [reply]
This was indeed the cause of this issue. Thanks a lot for your help!
